File: src/main/java/upbrella/be/store/repository/StoreDetailRepository.java
package upbrella.be.store.repository;

import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;
import upbrella.be.store.entity.StoreDetail;


public interface StoreDetailRepository extends JpaRepository<StoreDetail, Long>, StoreDetailRepositoryCustom {

    Optional<StoreDetail> findStoreDetailByStoreMetaId(long storeMetaId);
}

File: src/main/kotlin/upbrella/be/store/repository/StoreDetailRepository.kt
package upbrella.be.store.repository

import org.springframework.data.jpa.repository.JpaRepository
import upbrella.be.store.entity.StoreDetail
import java.util.*

interface StoreDetailRepository : JpaRepository<StoreDetail, Long>, StoreDetailRepositoryCustom {
    fun findStoreDetailByStoreMetaId(storeMetaId: Long): Optional<StoreDetail>
}